### ISSUE TITLE
[Stub Content] Plugin should not execute if vanilla app is not enabled

### ISSUE THREAD

**Issue by kaecyra** [vanilla/vanilla#7278]
When the core vanilla application is not enabled, stubcontent should not fire. Currently it does fire, and it depends on models provided by the vanilla application. This should be prevented. 

Stubcontent may be better as being bundled in the vanilla application and fired as a hook in its hooks file?

**Comment by R-J** [comment]
But isn't that what the PluginInfo key, `RequiredApplications` is made for?

Adding a plugins functionality to core which is used once during setup (and not even needed to be executed then) seems like adding unnecessary bloat to Vanilla

**Comment by kaecyra** [comment]
What we could do is add it to the vanilla app’s configuration.php in EnabledPlugins there. This doesn’t technically prevent conflicts or dynamically resolve the dependency but it gets some of the way there.

**Comment by linc** [comment]
Get @kaecyra's approval on the PR **after** it's gotten its double signoff.

**Comment by linc** [comment]
Acceptance criteria:

* Stub content must not enable if we opt out of installing Vanilla during setup.
* Stub content must still enable correctly if we do a default GUI setup OR default /utility/update setup.

**Comment by linc** [comment]
Suggestions:

1. Try Tim's approach and move it to Vanilla's (the app) config-defaults
2. Use logic in Vanilla (the app's) structure method in the hooks file.

**Comment by topcheese13** [comment]
We are blocked on this, the initial fix I tried (removing stubcontent from config-defaults and triggering the install from the `setup()` method of the vanilla settings hooks file, worked for the a GUI install. We ran into unit tests issues.

This test was failing https://github.com/vanilla/vanilla/blob/master/tests/APIv0/AltTest.php#L33

**Comment by topcheese13** [comment]
I tried multiple things.

1. Moving it to Vanilla's (the app) configuration. Didn't work. Stub isn't firing + unit tests failing.

2. Removing it from config-defaults and adding a saveToConfig stub to true in Vanilla's (the app) setup() method in its hook file. Didn't work.Unit test failing.

3. Setting Vanilla (the app) as a requirement in the addon.json on the stub and removing it from config-defaults. Didn't work. Stub isn't firing + unit tests failing.

4. Adding a check to the stub setup() method to confirm Vanilla (the app) is enabled. Works from the GUI. Works with a post request to dashboard/setup which is essentially the same thing as the GUI meuh. The stub doesn't fire when I set Vanilla (the app) to not be installed in the config-defaults (`$Configuration['Garden']['Install']['Applications'] = ['Conversations'];`), overwritting the default of `['Conversations', 'Vanilla']` in the setup controller. I still need to confirm if it works with a default /utility/update setup but I'm not sure what that entails or how to replicate/test it.

>Stub content must still enable correctly if we do a default GUI setup OR default /utility/update setup.
